passport_strategies/initLocalAuth.js

```diff
@@ -9,21 +9,25 @@ module.exports = function localAuthStrategy(passport) {
     User.findByEmail(email).then((user) => {
       if (!user) {
         console.log(`Email does not exist: ${email}`);
-        return done(null, false, {
+        done(null, false, {
           message: 'bad email',
         });
-      }
-      if (password !== user.password) {
-        console.log(`Incorrect password for user: ${user.email}`);
-        return done(null, false, {
-          message: 'bad password',
+      } else {
```

Comment:
I don't think we should have this as if/else. It should be if(!user || bad password) return "incorrect username or password"

so like:

User.comparePassword(password, user.password).then((success) => {
    if(!success || !user) reject;
    else 
        sucess
});